### PR TITLE
The event_loss calculator is now able to produce aggregate loss curves

### PR DESCRIPTION
**Pull request by micheles**
Another step towards the direction of replacing the event based risk calculator. See https://bugs.launchpad.net/oq-engine/+bug/1438217
The tests are green: https://ci.openquake.org/job/zdevel_oq-risklib/460

### CODE REVIEW

File: openquake/risklib/riskinput.py [hunk 1]
```diff
@@ -118,26 +118,32 @@ def build_input(self, imt, hazards_by_site, assets_by_site, eps_dict=None,
         return RiskInput(imt_taxonomies, hazards_by_site,
                          assets_by_site, eps_dict)
 
-    def build_inputs_from_ruptures(self, sitecol, assets_by_site, ses_ruptures,
-                                   gsims, trunc_level, correl_model, eps_dict,
-                                   epsilon_sampling):
+    def build_inputs_from_ruptures(self, sitecol, assets_by_site, all_ruptures,
+                                   gsims_by_trt_id, trunc_level, correl_model,
+                                   eps_dict, hint):
         """
```
Comment (ptormene) [inline openquake/risklib/riskinput.py]: Please update the docstring

File: openquake/commonlib/calculators/event_loss.py [hunk 2]
```diff
@@ -104,30 +105,29 @@ def pre_execute(self):
...
+        num_samples = min(len(all_ruptures), epsilon_sampling)
+        eps_dict = riskinput.make_eps_dict(
+            self.assets_by_site, num_samples,
+            getattr(oq, 'master_seed', 42),
+            getattr(oq, 'asset_correlation', 0))
+        logging.info('Generated %d epsilons', num_samples * len(eps_dict))
+        self.riskinputs = list(self.riskmodel.build_inputs_from_ruptures(
+            self.sitecol, self.assets_by_site, all_ruptures,
+            gsims_by_trt_id, oq.truncation_level, correl_model, eps_dict,
+            oq.concurrent_tasks // 2))
```
Comment (ptormene) [inline openquake/commonlib/calculators/event_loss.py]: Please add a comment about the division by 2

File: openquake/commonlib/calculators/event_loss.py [hunk 3]
```diff
@@ -137,19 +137,61 @@ def post_execute(self, result):
...
-                             ordinal, loss_type, nonzero, total)
+
+                # aggregate loss curves per asset
+                losses_by_asset = groupby(  # (tag, asset_id, loss) triples
+                    ela, operator.itemgetter(1),
+                    lambda rows: [row[2] for row in rows])
+                key = 'rlz-%03d-%s-loss-curves' % (ordinal, loss_type)
+                self.risk_out[key] = []
+                for asset_id, losses in losses_by_asset.iteritems():
+                    losses, poes, avg, std = self.build_loss_curve(losses)
```
Comment (ptormene) [inline openquake/commonlib/calculators/event_loss.py]: Please avoid the ambiguity of `losses`

File: openquake/baselib/general.py [hunk 4]
```diff
@@ -218,6 +218,51 @@ def split_in_blocks(sequence, hint, weight=lambda item: 1,
...
+    [0, 1, 2] ['A']
+    [3, 4, 5] ['B']
+    [6, 7, 8] ['C']
+    [9] ['A']
+    """
+    N, n = len(long_sequence), len(short_sequence)
+    assert N >= n
+    long_blocks = split_in_blocks(long_sequence, hint, weight, key)
+    short_blocks = split_in_blocks(short_sequence, hint)
+    for long, short in zip(long_blocks, itertools.cycle(short_blocks)):
```
Comment (ptormene) [inline openquake/baselib/general.py]: Please rename `long`

File: openquake/risklib/riskinput.py (hunk 1 above)
Comment (micheles) [inline openquake/risklib/riskinput.py]: Done

File: openquake/commonlib/calculators/event_loss.py (hunk 2 above)
Comment (micheles) [inline openquake/commonlib/calculators/event_loss.py]: Added

File: openquake/commonlib/calculators/event_loss.py (hunk 3 above)
Comment (micheles) [inline openquake/commonlib/calculators/event_loss.py]: Done

File: openquake/baselib/general.py (hunk 4 above)
Comment (micheles) [inline openquake/baselib/general.py]: Removed

File: openquake/commonlib/commands/info.py [hunk 5]
```diff
@@ -32,8 +32,16 @@ def info(name, filtersources=False):
         print textwrap.dedent(base.calculators[name].__doc__.strip())
     elif name.endswith('.ini'):
         oqparam = readinput.get_oqparam(name)
-        sitecol = (readinput.get_site_collection(oqparam)
-                   if filtersources else None)
+        if filtersources:
+            if 'exposure' in oqparam.inputs:
+                expo = readinput.get_exposure(oqparam)
```
Comment (ptormene) [inline openquake/commonlib/commands/info.py]: `expo` might not be ready on time

**Comment by ptormene** [comment]
Besides the beautiful joke, LGTM

**Pull request merged**